ODIN_II/SRC/netlist_create_from_ast.cpp

```diff
@@ -3412,8 +3415,12 @@ signal_list_t* create_operation_node(ast_node_t* op, signal_list_t** input_lists
         case ASL: // <<<
         case SL:  // <<
             /* Shifts doesn't matter about port size, but second input needs to be a number */
-            //output_port_width = input_lists[0]->count + (shift_left_value_with_overflow_check(0x1, input_lists[1]->count)-1);
-            output_port_width = input_lists[0]->count + (shift_left_value_with_overflow_check(0x1, log2(op->children[1]->types.vnumber->get_value()), op->loc));
+            // TEMPERORY ASSIGNMENT USING IF EXPRESSION
+            if (op->children[1]->type != NUMBERS)
+                output_port_width = input_lists[0]->count + (shift_left_value_with_overflow_check(0x1, input_lists[1]->count, op->loc)-1);
+            else
```

Comment:
Don't use this. Numbers should never get here. Ever.
If they do you have a bug

Reply:
any chance you can figure out how to grab the real output width? the whole thing is recursive, just pass the assignment width just like the resolve number function

Reply:
As you said since the function is completely recursively, we do everything related to output completely before doing BINARY_OPERATION like the shift. For instance:
```
left_shift = a_in << 2;
```
We have a BLOCKING_STATEMENT, containing IDENTIFIER (left_shift) and BINARY_OPERATION (a_in << 2). So, when we create pins related to a_in (based on the size it reads from its indexes) we do not have access left_shift size! Please look at line 491 of netlist_create_from_ast, at this place we specify and create the pins of each variable. So, we are unable to reach the sibling of the current node! Please update me with your opinion!

Reply:
You have to do it during ast elaboration at the last phase when it resolve the assignment size, you have to resize the inputs as I said before...

Reply:
whats the ETA on resolving this comment?